core/src/main/java/com/airbnb/aerosolve/core/util/ReinforcementLearning.java

```diff
@@ -0,0 +1,60 @@
+package com.airbnb.aerosolve.core.util;
+
+/**
+ * Utilities for reinforcement learning
+ */
+
+import com.airbnb.aerosolve.core.models.AbstractModel;
+import com.airbnb.aerosolve.core.FeatureVector;
+import com.google.common.hash.HashCode;
+import com.google.common.hash.Hasher;
+import com.google.common.hash.Hashing;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.thrift.TDeserializer;
+import org.apache.thrift.TSerializer;
+import org.apache.thrift.TBase;
+
+import java.io.Serializable;
+import java.util.*;
+
+public class ReinforcementLearning implements Serializable {
+  // Updates a model using SARSA
+  // https://en.wikipedia.org/wiki/State-Action-Reward-State-Action
+  // If the nextState is terminal, pass in null
+  public static void updateSARSA(AbstractModel model,
+                                 FeatureVector stateAction,
+                                 float reward,
+                                 FeatureVector nextStateAction,
+                                 float learningRate,
+                                 float decay) {
```

Comment:
I would call this "discountRate" or "discountFactor" to match the terminology in the literature.